BUG: empty gcps with older versions of rasterio (1.2.1)

See: https://github.com/corteva/rioxarray/actions/runs/1819395031

```
E           TypeError: Invalid value for attr 'gcps': {'type': 'FeatureCollection', 'features': []}. For serialization to netCDF files, its value must be of one of the following types: str, Number, ndarray, number, list, tuple
```

--- a/rioxarray/_io.py
+++ b/rioxarray/_io.py
@@ -880,7 +880,7 @@ def open_rasterio(
         coord_name = "band"
         coords[coord_name] = np.asarray(riods.indexes)
 
-    has_gcps = hasattr(riods, "gcps") and riods.gcps != ([], None)
+    has_gcps = hasattr(riods, "gcps") and riods.gcps[0] != []
     if has_gcps:
         parse_coordinates = False
 

--- a/rioxarray/_io.py
+++ b/rioxarray/_io.py
@@ -880,7 +880,7 @@ def open_rasterio(
         coord_name = "band"
         coords[coord_name] = np.asarray(riods.indexes)
 
-    has_gcps = hasattr(riods, "gcps") and riods.gcps[0] != []
+    has_gcps = hasattr(riods, "gcps") and riods.gcps[0]
     if has_gcps:
         parse_coordinates = False
 

